### DISCUSSION TITLE
unicode parsing issues for sample data provided?

### DISCUSSION THREAD

**Discussion by fractaloutlook** [Blockchain-Technology-Lab/consensus-decentralization#156] [Q&A] [answered]
Heyo! Not sure if this project is still active, but:

When running it, 1) - the output messaging seems to claim that it is ignoring the ledgers I asked for... and 2) the execution end with an error about not being able to decode a character to unicode format?

It has that same error with or without the ledgers parameter, or using any other parameters. I downloaded the sample data for bitcoin, tezos, cardano, and ethereum from the page linked in the readme.

![2024-06-23 18_03_38-Window](https://github.com/Blockchain-Technology-Lab/consensus-decentralization/assets/27757547/a93d4586-8699-4699-b1bd-0c562c1a0bb7)

Any thoughts on this one?

**Comment by LadyChristina** [comment]
Hello @fractaloutlook, thanks for raising this to our attention. 

For your first point, we very recently changed the way parameters are specified and I only now realise that we forgot to update the documentation, sorry about that! We will update the docs asap, but basically all the configurations are now done through the [config.yaml](https://github.com/Blockchain-Technology-Lab/consensus-decentralization/blob/main/config.yaml) file. So if you want only one ledger to be included, you would need to comment out the rest. Let me know if you have any questions on this.

For your second issue, I'm not sure why this is happening, it's not something I can reproduce right now. Can you confirm that you haven't made any changes to [this file](https://github.com/Blockchain-Technology-Lab/consensus-decentralization/blob/main/mapping_information/identifiers/bitcoin.json) on your local copy?

**Reply by fractaloutlook** [reply]
Yep, my file is the same, I never made any changes to it. I've got my yaml file pared down to just the 4 I have data for.... and it still failed at the same point. I then commented out bitcoin from the list, and Got this different error:

![2024-06-24 19_10_09-Window](https://github.com/Blockchain-Technology-Lab/consensus-decentralization/assets/27757547/677a6644-1c15-4ca7-bf8a-23710a54f123)

I was running cardano, ethereum, tezos. ran the py file with no parameters.

**Reply by LadyChristina** [reply, marked as answer]
Thanks for the response. I think I figured out the issue - we have to explicitly specify the encoding when reading the identifier files in order for them to be parsed on Windows. I have opened a [PR](https://github.com/Blockchain-Technology-Lab/consensus-decentralization/pull/159) that solves this issue, I will let you know once it is approved and merged into the main branch

**Reply by fractaloutlook** [reply]
Great thanks! Would the instructions and files as-they-are have worked in WSL Ubuntu? Just curious.

**Reply by LadyChristina** [reply]
Yes, they should work in WSL as they are

**Reply by LadyChristina** [reply]
Fyi the PR has now been merged so if you pull you should be able to run the code on Windows too (but do let us know if any problems persist!)
